rust/src/builder.rs

```diff
@@ -111,6 +127,7 @@ impl DeltaTableLoadOptions {
             storage_backend: None,
             require_tombstones: true,
             require_files: true,
+            log_buffer_size: 1,
```

Comment:
I'm not sure we want to default to the least performant option here. I think I'd prefer:

```suggestion
            log_buffer_size: num_cpus::get() * 4,
```

Usually the memory allotted is proportional to number of CPUs, as is the capability of the machine to handle many concurrent requests, so it's a good heuristic.